File: 724. Find Pivot Index/java/myans.java
public class myans {
    public int pivotIndex(int[] nums) {
        int sumOfList = 0;
        for (int i = 0; i < nums.length; i++) {
            sumOfList = sumOfList + nums[i];
        }
        int leftsum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (leftsum == (sumOfList - leftsum - nums[i]))
                return i;
            leftsum = leftsum + nums[i];
        }
        return -1;
    }
}

File: 724. Find Pivot Index/python/myans.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        sum_of_list = sum(nums)
        leftsum = 0
        for index, num in enumerate(nums):
            if leftsum == (sum_of_list - leftsum - num):
                return index
            leftsum = leftsum + nums[index]
        return -1


nums = [1, 2, 3]

output = Solution().pivotIndex(nums)
print(output)
